### ISSUE TITLE
snapwidget.cpp insists on having zxing/common/GlobalHistogramBinarizer.h even though ZXING was not detected by configure

### ISSUE THREAD

**Issue by allesklaar** [FeatherCoin/Feathercoin#176]
This is 0.9.6.1 branch, but release tag v0.9.6 has the same issue.

I don't have zxing installed and your configure does not detect it:

```./configure --with-incompatible-bdb```
```
$ grep -i 'xing' config.log config.status 
config.log:configure:27088: checking for main in -lzxing
config.log:configure:27107: g++ -std=c++11 -o conftest -g -O2 -Wall -Wextra -Wformat -Wformat-security -Wno-unused-parameter  -Wstack-protector -fstack-protector-all -fPIE -fPIC  -DBOOST_SPIRIT_THREADSAFE -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2   -Wl,-z,relro -Wl,-z,now -pie conftest.cpp -lzxing  -lminiupnpc -lminiupnpc -lminiupnpc -lminiupnpc  -lssl -lcrypto >&5
config.log:/usr/bin/ld: cannot find -lzxing
config.log:ac_cv_lib_zxing_main=no
config.log:USE_ZXING=''
config.status:S["USE_ZXING"]=""
```

Unexpectedly build fails complaining about the missing zxing includes:

```
In file included from walletview.cpp:23:0:
utilitydialog.h: At global scope:
utilitydialog.h:30:20: warning: 'Ui::OPENNAME_NAME_RENEWAL' defined but not used [-Wunused-variable]
 static const char* OPENNAME_NAME_RENEWAL="e";
                    ^~~~~~~~~~~~~~~~~~~~~
utilitydialog.h:29:20: warning: 'Ui::OPENNAME_NAME_TRANSFER' defined but not used [-Wunused-variable]
 static const char* OPENNAME_NAME_TRANSFER="d";
                    ^~~~~~~~~~~~~~~~~~~~~~
utilitydialog.h:28:20: warning: 'Ui::OPENNAME_NAME_UPDATE' defined but not used [-Wunused-variable]
 static const char* OPENNAME_NAME_UPDATE="c";
                    ^~~~~~~~~~~~~~~~~~~~
utilitydialog.h:27:20: warning: 'Ui::OPENNAME_NAME_REGISTRATION' defined but not used [-Wunused-variable]
 static const char* OPENNAME_NAME_REGISTRATION="b";
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
utilitydialog.h:26:20: warning: 'Ui::OPENNAME_NAME_PREORDER' defined but not used [-Wunused-variable]
 static const char* OPENNAME_NAME_PREORDER="a";
                    ^~~~~~~~~~~~~~~~~~~~~~
  CXX      libbitcoinqt_a-reportview.o
In file included from reportview.cpp:13:0:
walletmodel.h:11:0: warning: ignoring #pragma warning  [-Wunknown-pragmas]
 #pragma warning(disable:4717)  //bogus warning from MS
 
  CXX      libbitcoinqt_a-snapwidget.o
snapwidget.cpp:10:51: fatal error: zxing/common/GlobalHistogramBinarizer.h: No such file or directory
 #include "zxing/common/GlobalHistogramBinarizer.h"
                                                   ^
compilation terminated.
make[4]: *** [Makefile:2126: libbitcoinqt_a-snapwidget.o] Error 1
make[4]: Leaving directory '/home/data/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[3]: *** [Makefile:3018: all-recursive] Error 1
make[3]: Leaving directory '/home/data/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[2]: *** [Makefile:1164: all] Error 2
make[2]: Leaving directory '/home/data/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[1]: *** [Makefile:993: all-recursive] Error 1
make[1]: Leaving directory '/home/data/software/gnu_linux/feathercoin/Feathercoin/src'
make: *** [Makefile:579: all-recursive] Error 1
```

I don't want zxing and QR code support. Your configure script appears to allow for that, but the build process does not honor it.

**Comment by wellenreiter01** [comment]
This is a known problem.

You can run 'confgure --with-incompatible-bdb --with-qrencode=no' to avoid the problem.

We are working on a fix

**Comment by allesklaar** [comment]
```--with-qrencode=no`` did not help as of latest commit on branch 0.9.6.1 when I opened this ticket. Tried again on the latest #b5d8bc05b7711cae7db1ead9ec4e75fbcaee1d78 just now and build still fails in the same manner.

**Comment by wrapperband** [comment]
Can you give us the error massages and some info on the operating system, we tested 0.9.6.1 build many times with and without qr codes.

**Comment by allesklaar** [comment]
@wrapperband 

You have the error message right there in this ticket. It's make output.

I suppose that even though you set ```--with-qrencode-no``` you still have ZXING installed on your system, which lets the build find GlobalHistogramBinarizer.h on your system and pass, even though it doesn't actually compile the code that depends on ZXING. Try uninstalling ZXING completely and then build. Does make pass then?

**Comment by wellenreiter01** [comment]
@allesklaar: Can you pull the latest commit of 0.9.6.1 and try again?

**Comment by allesklaar** [comment]
Same error in 0.9.6.1 e977bc52740507636c73a8e5c7768665acc16fda.

```
./autogen.sh
./configure --with-qrencode=no --with-incompatible-bdb
make
```
```
  CXX      libbitcoinqt_a-snapwidget.o
snapwidget.cpp:10:51: fatal error: zxing/common/GlobalHistogramBinarizer.h: No such file or directory
 #include "zxing/common/GlobalHistogramBinarizer.h"
                                                   ^
compilation terminated.
make[4]: *** [Makefile:2126: libbitcoinqt_a-snapwidget.o] Error 1
make[4]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[3]: *** [Makefile:3018: all-recursive] Error 1
make[3]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[2]: *** [Makefile:1164: all] Error 2
make[2]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[1]: *** [Makefile:993: all-recursive] Error 1
make[1]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src'
make: *** [Makefile:579: all-recursive] Error 1
```

**Comment by aciddude** [comment]
Hi @allesklaar 

can you try this: 

```
./autogen.sh
./configure --with-qrcode=no --with-incompatible-bdb
make
```

**Comment by wellenreiter01** [comment]
@aciddude 
That will not solve the problem. The problem is caused by the handling of the compiler variables USE_QRCODE and USE_ZXING
I think, I have a fix, but need to verify that my fix is working well with ./configure --with-qrcode=no and ./configure --with-qrcode=yes

**Comment by allesklaar** [comment]
@natedavis, @wellenreiter01 

The build is still failing in a , as of b7b59064287d6912f8df2dfbde201df1797f6a09 on branch 0.9.6.1:

```
./autogen.sh
./configure --with-qrcode=no --with-incompatible-bdb
make
```
```
  CXX      libbitcoinqt_a-snapwidget.o
snapwidget.cpp:10:51: fatal error: zxing/common/GlobalHistogramBinarizer.h: No such file or directory
 #include "zxing/common/GlobalHistogramBinarizer.h"
                                                   ^
compilation terminated.
make[5]: *** [Makefile:2056: libbitcoinqt_a-snapwidget.o] Error 1
make[5]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[4]: *** [Makefile:2926: all-recursive] Error 1
make[4]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[3]: *** [Makefile:1109: all] Error 2
make[3]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src/qt'
make[2]: *** [Makefile:922: all-recursive] Error 1
make[2]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src'
make[1]: *** [Makefile:676: all] Error 2
make[1]: Leaving directory '/home/dane/software/gnu_linux/feathercoin/Feathercoin/src'
make: *** [Makefile:511: all-recursive] Error 1
```

**Comment by FeatherCoin** [comment]
solved in latest code in master-0.9 and branch 0.9.6.2

**Comment by wellenreiter01** [comment]
fixed in 0.9.6.2
